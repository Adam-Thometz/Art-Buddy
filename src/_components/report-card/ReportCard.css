.ReportCard {
  width: 90%;
  margin: 0 auto;
  padding: 0 50px;
}

.ReportCard-scores {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-template-rows: repeat(3, 1fr);
}

.ReportCard-clear {
  padding-top: 40px;
  margin: 0 auto;
  text-decoration: underline;
  font-size: 1.5rem;
}