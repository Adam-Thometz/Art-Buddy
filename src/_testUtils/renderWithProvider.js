import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import { MemoryRouter } from 'react-router-dom';

// import renderWithRouter from './renderWithRouter';
import rootReducer from '../_redux/reducers/rootReducer';

function withProvider({ children, store }) {
  return (
    <Provider store={store}>
      <MemoryRouter>
        {children}
      </MemoryRouter>
    </Provider>
  )
};

function renderWithProvider(component, options) {
  const store = (options && options.store)
    ? options.store
    : rootReducer;
  return render(component, { wrapper: (args) =>
    withProvider({
      ...args,
      store
    }), ...options
  });
};

export default renderWithProvider;