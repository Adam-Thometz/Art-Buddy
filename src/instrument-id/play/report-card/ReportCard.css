.ReportCard {
  width: 90%;
  margin: 0 auto;
  padding: 0 50px;
}

.ReportCard-scores {
  display: grid;
  grid-template-columns: repeat(2, 1fr);
  grid-template-rows: repeat(3, 1fr);
}

.ReportCard-score-display {
  font-size: 2rem;
  margin: 5% 0;
  text-align: left;
  display: flex;
  flex-direction: column;
}

.ReportCard-visual-score img {
  padding-right: 10px;
  height: 50px;
  width: 23%;
}

.ReportCard-text-score {
  text-align: right;
  font-size: 1.5rem;
  margin-right: 5%;
}

.ReportCard-clear {
  padding-top: 40px;
  margin: 0 auto;
  text-decoration: underline;
  font-size: 1.5rem;
}