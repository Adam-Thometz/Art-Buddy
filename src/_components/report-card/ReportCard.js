import useReportCard from '_hooks/report-card/useReportCard';

import './ReportCard.css';

import Score from './score/Score';

import * as reportCards from '_data/_report-cards/initialReportCards';

const ReportCard = ({ game, level = null }) => {
  const [reportCard, setReportCard] = useReportCard(game, level);
  const scoreDisplay = Object.keys(reportCard).map(group => {
    const { name, results, maxScore, colorPalette } = reportCard[group];
    return <Score key={name} name={name} score={results.length} maxScore={maxScore} colorPalette={colorPalette} />;
  });

  const handleClearReportCard = () => setReportCard(reportCards[`${game}ReportCard`]);

  return (
    <main className='ReportCard'>
      <section className='ReportCard-scores'>
        {scoreDisplay}
      </section>
      <p className='ReportCard-clear' onClick={handleClearReportCard}>clear report card</p>
    </main>
  );
};

export default ReportCard;