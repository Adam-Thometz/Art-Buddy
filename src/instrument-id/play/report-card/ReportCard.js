import React from 'react';
import Button from '../../../_components/button/Button';
import useReportCard from '../../../_hooks/useReportCard';

import './ReportCard.css';

import { INITIAL_REPORT_CARD } from '../../../_hooks/useReportCard';
import { emptyFirst, emptyMid, emptyLast, fullFirst, fullMid, fullLast } from './_icons/iconImports';

const ReportCard = ({ level }) => {
  const [savedReportCard, setSavedReportCard] = useReportCard(level);

  const scoreDisplay = Object.keys(savedReportCard).map(family => {
    const score = savedReportCard[family].length;
    const maxScore = 4;
    return (
      <section className='ReportCard-score-display'>
        <span>{family.toUpperCase()} FAMILY</span>
        <div className='ReportCard-visual-score'>
          <img src={score >= 1 ? fullFirst : emptyFirst} alt='' />
          <img src={score >= 2 ? fullMid : emptyMid} alt='' />
          <img src={score >= 3 ? fullMid : emptyMid} alt='' />
          <img src={score >= maxScore ? fullLast : emptyLast} alt='' />
        </div>
        <span className='ReportCard-text-score'>{score}/{maxScore} passed</span>
      </section>
    );
  });

  const handleClearReportCard = () => {
    setSavedReportCard(JSON.stringify(INITIAL_REPORT_CARD));
  };

  return (
    <main className='ReportCard'>
      {scoreDisplay}
      <Button colorId={2} onClick={handleClearReportCard}>CLEAR</Button>
    </main>
  );
}

export default ReportCard;