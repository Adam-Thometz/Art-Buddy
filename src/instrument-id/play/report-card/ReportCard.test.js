import React from "react";
import renderWithProvider from "_testUtils/renderWithProvider";
import ReportCard from "./ReportCard";

describe('ReportCard component', () => {
  it('renders without crashing', () => {
    renderWithProvider(<ReportCard />);
  });

  it('matches snapshot', () => {
    const { asFragment } = renderWithProvider(<ReportCard />);
    expect(asFragment()).toMatchSnapshot();
  });
});