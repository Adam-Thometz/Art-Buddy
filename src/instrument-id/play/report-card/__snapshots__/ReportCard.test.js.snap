// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReportCard component matches snapshot 1`] = `
<DocumentFragment>
  <main
    class="ReportCard"
  >
    <section
      class="ReportCard-score-display"
    >
      <span>
        BRASS FAMILY
      </span>
      <div
        class="ReportCard-visual-score"
      >
        <img
          alt=""
          src="file"
        />
        <img
          alt=""
          src="file"
        />
        <img
          alt=""
          src="file"
        />
        <img
          alt=""
          src="file"
        />
      </div>
      <span
        class="ReportCard-text-score"
      >
        0/4 passed
      </span>
    </section>
    <section
      class="ReportCard-score-display"
    >
      <span>
        ELECTRONIC FAMILY
      </span>
      <div
        class="ReportCard-visual-score"
      >
        <img
          alt=""
          src="file"
        />
        <img
          alt=""
          src="file"
        />
        <img
          alt=""
          src="file"
        />
        <img
          alt=""
          src="file"
        />
      </div>
      <span
        class="ReportCard-text-score"
      >
        0/4 passed
      </span>
    </section>
    <section
      class="ReportCard-score-display"
    >
      <span>
        PERCUSSION FAMILY
      </span>
      <div
        class="ReportCard-visual-score"
      >
        <img
          alt=""
          src="file"
        />
        <img
          alt=""
          src="file"
        />
        <img
          alt=""
          src="file"
        />
        <img
          alt=""
          src="file"
        />
      </div>
      <span
        class="ReportCard-text-score"
      >
        0/4 passed
      </span>
    </section>
    <section
      class="ReportCard-score-display"
    >
      <span>
        STRINGS FAMILY
      </span>
      <div
        class="ReportCard-visual-score"
      >
        <img
          alt=""
          src="file"
        />
        <img
          alt=""
          src="file"
        />
        <img
          alt=""
          src="file"
        />
        <img
          alt=""
          src="file"
        />
      </div>
      <span
        class="ReportCard-text-score"
      >
        0/4 passed
      </span>
    </section>
    <section
      class="ReportCard-score-display"
    >
      <span>
        WOODWIND FAMILY
      </span>
      <div
        class="ReportCard-visual-score"
      >
        <img
          alt=""
          src="file"
        />
        <img
          alt=""
          src="file"
        />
        <img
          alt=""
          src="file"
        />
        <img
          alt=""
          src="file"
        />
      </div>
      <span
        class="ReportCard-text-score"
      >
        0/4 passed
      </span>
    </section>
    <section
      class="ReportCard-score-display"
    >
      <span>
        VOICE FAMILY
      </span>
      <div
        class="ReportCard-visual-score"
      >
        <img
          alt=""
          src="file"
        />
        <img
          alt=""
          src="file"
        />
        <img
          alt=""
          src="file"
        />
        <img
          alt=""
          src="file"
        />
      </div>
      <span
        class="ReportCard-text-score"
      >
        0/4 passed
      </span>
    </section>
    <button
      class="Button regular"
      style="border-color: #fd4d3a;"
    >
      CLEAR
    </button>
  </main>
</DocumentFragment>
`;
